tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -809,3 +814,27 @@ size_t lrs_device_controller::get_index_of_profile( const realdds::dds_stream_pr
 
     return 0;
 }
+
+
+bool lrs_device_controller::on_control( std::string const & id, nlohmann::json const & control, nlohmann::json & reply )
+{
+    static std::map< std::string, bool ( lrs_device_controller::* )( nlohmann::json const &, nlohmann::json & ) > const
+        control_handlers{
+            { "hw-reset", &lrs_device_controller::on_hardware_reset },
+            { "open-streams", &lrs_device_controller::on_open_streams },
+        };
+    auto it = control_handlers.find( id );
+    if( it == control_handlers.end() )
+        return false;
+
+    return (this->*(it->second))( control, reply );
+}
+
+
+bool lrs_device_controller::on_hardware_reset( nlohmann::json const & control, nlohmann::json & reply )
+{
+    _rs_dev.hardware_reset();
+    return true;
```

Comment:
No need to set reply to OK?

Reply:
If we do not throw anything, it's OK. The `true` means we handled it.